src/redux/modules/broker.js

```diff
@@ -0,0 +1,67 @@
+import { handleActions } from 'redux-actions';
+import { createRecord, resourceStatus } from '../helpers/resourceManager';
+import { createApiAction } from '../middleware/apiMiddleware';
+import { fromJS } from 'immutable';
+
+export const actionTypes = {
+  BROKER_STATS_FETCH: 'recodex/broker/STATS_FETCH',
+  BROKER_STATS_FETCH_PENDING: 'recodex/broker/STATS_FETCH_PENDING',
+  BROKER_STATS_FETCH_FULFILLED: 'recodex/broker/STATS_FETCH_FULFILLED',
+  BROKER_STATS_FETCH_REJECTED: 'recodex/broker/STATS_FETCH_REJECTED',
+  BROKER_FREEZE: 'recodex/broker/FREEZE',
+  BROKER_UNFREEZE: 'recodex/broker/UNFREEZE'
```

Comment:
I am not really fond of this setup. Freeze/Unfreeze action should monitor their progress (i.e., pending, fulfilled and rejected actions should also have their reducers and the pending/failed status should be recorded in redux, so it can be displayed on the page (and possibly used to prevent sending concurrent actions form UI).